tests/simulator_test.cpp

```diff
@@ -2,82 +2,217 @@
+  ASSERT_EQ(performHitCollector(energyCollectors, straightUp, &hitData), true);
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+
+  Vec3 collisionPointStraightUp =
+      Vec3(0, 0, collectorPositionRadius - refCollectorRadius);
+  ASSERT_EQ(hitData.collisionPoint(), collisionPointStraightUp);
+
+  Ray straightDown(kVecZero, -kVecUp);
+  ASSERT_FALSE(performHitCollector(energyCollectors, straightDown, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray alongX(kVecZero, kVecX);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
+
+  ASSERT_EQ(alongX.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray alongY(kVecZero, kVecY);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
+
+  ASSERT_EQ(alongY.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray atSixty(kVecZero, Vec3(std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixty, &hitData));
+
+  Ray atSixtyOther(kVecZero,
+                   Vec3(-std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixtyOther, &hitData));
+
+  Ray atSixtyXY(kVecZero, Vec3(std::cos(deg2rad(60)), std::cos(deg2rad(60)),
+                               std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXY, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray atSixtyXYOther(kVecZero,
+                     Vec3(-std::cos(deg2rad(60)), -std::cos(deg2rad(60)),
+                          std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXYOther, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+}
+
+TEST_F(EnergyCollectorTest, EvenNumOfEnergyCollectorTest) {
+  const FakeModel nonEmptyModel(false);
+
+  const int numCollectors = 36;
+  auto energyCollectors = buildCollectors(nonEmptyModel, numCollectors);
+  ASSERT_EQ(energyCollectors.size(), numCollectors);
+
+  Ray straightUp(kVecZero, kVecUp);
+  RayHitData hitData;
+  ASSERT_TRUE(performHitCollector(energyCollectors, straightUp, &hitData));
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 2);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+
+  Vec3 OriginOfPenultimateCollector =
+      energyCollectors[energyCollectors.size() - 2]->getOrigin();
+  Vec3 OriginOfLastCollector =
+      energyCollectors[energyCollectors.size() - 1]->getOrigin();
+  Vec3 refCollision =
+      (OriginOfLastCollector + OriginOfPenultimateCollector) / 2 -
+      Vec3(0, 0, refCollectorRadius * std::sqrt(3) / 2);
```

Comment:
Where is this coming from?

Reply:
added comment about it ;-) I read in the book that quality of code can be measured by number of wtf per minute of reading ;P so I will try to get as less of wtfs as possible from now on :-P